src/pieces.rs

```diff
@@ -17,11 +17,11 @@ pub enum TowerHeight {
     Empty,
 }
 
-impl<'a> Tower<'a> {
+impl Tower {
     /// Returns the top most piece and a tower that has its top piece removed
     /// Returns Err if the tower is empty
     /// This function does not modify the original tower
```

Comment:
This line of documentation is maybe unnecessary, since it's a `&self` method and because the description suggests as much by "Returns the top most piece and a tower that has its top piece removed" implying the fact that it's a pure function.